js/MVP/hooks/brev.js

```diff
@@ -0,0 +1,32 @@
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { getBrev, postLestBrev } from '../services/brev';
+
+const MOTEINNKALLELSE = 'moteinnkallelse';
+
+export const useBrev = () => {
+  return useQuery(MOTEINNKALLELSE, getBrev, {
+    retry: 0,
+  });
+};
+
+const setLestDatoForVarsel = (uuid) => {
+  return (varsel) => {
+    if (varsel.uuid === uuid) {
+      return { ...varsel, lestDato: true };
+    }
+
+    return varsel;
+  };
+};
+
+export const useMutatebrevLest = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation(({ uuid }) => postLestBrev(uuid), {
+    onSuccess: (_data, variables) => {
```

Comment:
Hm, er tanken her optimistisk oppdatering av "lestDato"? I så fall kan det være lurt å invalidere query for å få ferske oppdaterte data fra backend. Ala 

```
onSettled: () => {
      queryClient.invalidateQueries(MOTEINNKALLELSE);
    },
```
    
ref https://react-query.tanstack.com/guides/optimistic-updates